src/drivers/uavcan/actuators/esc.hpp

```diff
@@ -94,13 +102,30 @@ class UavcanEscController
 	typedef uavcan::MethodBinder<UavcanEscController *,
 		void (UavcanEscController::*)(const uavcan::ReceivedDataStructure<uavcan::equipment::esc::Status>&)> StatusCbBinder;
 
+	typedef uavcan::MethodBinder<UavcanEscController *,
+		void (UavcanEscController::*)(const uavcan::ReceivedDataStructure<uavcan::equipment::esc::StatusExtended>&)>
+		StatusExtendedCbBinder;
+
 	typedef uavcan::MethodBinder<UavcanEscController *,
 		void (UavcanEscController::*)(const uavcan::TimerEvent &)> TimerCbBinder;
 
 	esc_status_s	_esc_status{};
 
 	uORB::PublicationMulti<esc_status_s> _esc_status_pub{ORB_ID(esc_status)};
 
+	//This is a PublicationMulti where we know all of the instances live here. So, let's make an array to store as many as we know how to handle (dronecan_esc_status_extended_s::CONNECTED_ESC_MAX)
+	uORB::PublicationMulti<dronecan_esc_status_extended_s>
```

Comment:
Was trying to figure out the best way to have a publication multi that all lives in one module. I came across the uORB unit testing that does exactly what I wanted (https://github.com/PX4/PX4-Autopilot/blob/d8be547b0694cfb1d7def95c5d894274f4748c5e/platforms/common/uORB/uORB_tests/uORBTest_UnitTest.cpp#L724C37-L724C55)

Admittedly, this seems a bit clunky. If this isn't what you're looking for, what would be your suggestion moving forward @dakejahl? I could do something similar to how you have the node status publication set up?

Reply:
yes that is what I'm thinking. I probably should have used the PublicationMulti abstraction in my PR, I just based it off of what already existed for the CanInterfaceStatus

Reply:
Sorry, just want to make sure we're on the same page with, "Yes that is what I'm thinking." One, that what I have in there now is how it should be, or two, that I should update this to match more closely what we do for CanInterfaceStatus?

Reply:
oh sorry I didn't realize you had pushed a commit. Yes this looks good :+1: